test/support/utils.rb

```diff
@@ -14,39 +14,66 @@ def create_message fail_msg
       "#{fail_msg}\n[DEBUG SESSION LOG]\n> " + @backlog.join('> ')
     end
 
-
     def combine_regexps(regexps)
       Regexp.new(regexps.map(&:source).reduce(:+))
     end
 
     def debug_code(program, **options, &block)
       @queue = Queue.new
-      block.call
+      @scenario = block
+      @scenario.call
       write_temp_file(strip_line_num(program))
-      create_pseudo_terminal(**options)
+      setup_terminal(**options)
       check_line_num!(program)
     end
 
     DEBUG_MODE = false
+    ASK_CMD = %w[quit delete kill]
 
     def debug_print msg
       print msg if DEBUG_MODE
     end
 
     RUBY = RbConfig.ruby
-    REPL_RPOMPT = /\(rdbg\)/
 
-    def create_pseudo_terminal(boot_options: "-r debug/run")
+    def setup_terminal(boot_options: "-r debug/run", remote: true)
+      ENV['RUBY_DEBUG_LOCAL_ONLY_MODE'] = 'false' # for fast debug test
       inject_lib_to_load_path
+
+      if remote && ENV['RUBY_DEBUG_LOCAL_ONLY_MODE'] == 'false'
```

Comment:
`RUBY_DEBUG_TEST` prefix is better to make clear the purpose.

Reply:
https://github.com/ruby/debug/pull/76/files#diff-ee14bf65f14927f7778a246f90c5cec0c581be3e2658c2a8a3726c59db5be3f9R41